services/skus/datastore.go

```diff
@@ -950,24 +950,40 @@ type TimeAwareSubIssuedCreds struct {
 	RequestID    string                    `json:"-" db:"request_id"`
 }
 
-func (pg *Postgres) AreTimeLimitedV2CredsSubmitted(ctx context.Context, blindedCreds ...string) (bool, error) {
+type AreTimeLimitedV2CredsSubmittedResult struct {
+	AlreadySubmitted bool `db:"already_submitted"`
+	Mismatch         bool `db:"mismatch"`
+}
+
+func (pg *Postgres) AreTimeLimitedV2CredsSubmitted(ctx context.Context, requestID uuid.UUID, blindedCreds ...string) (*AreTimeLimitedV2CredsSubmittedResult, error) {
+	return areTimeLimitedV2CredsSubmitted(ctx, pg.RawDB(), requestID, blindedCreds...)
+}
+
+type getContext interface {
+	GetContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+}
+
+func areTimeLimitedV2CredsSubmitted(ctx context.Context, dbi getContext, requestID uuid.UUID, blindedCreds ...string) (*AreTimeLimitedV2CredsSubmittedResult, error) {
 	if len(blindedCreds) < 1 {
-		return false, errors.New("invalid parameter to tlv2 creds signed")
+		return nil, errors.New("invalid parameter to tlv2 creds signed")
 	}
 
+	var result = AreTimeLimitedV2CredsSubmittedResult{}
+
 	query := `
 		select exists(
 			select 1 from time_limited_v2_order_creds where blinded_creds->>0 = $1
-		)
+		) as already_submitted,
+		exists(
+			select 1 from time_limited_v2_order_creds where blinded_creds->>0 = $1 and request_id != $2
```

Comment:
question: is it okay to only check the first value? For example, will the submitted creds i.e. `blindedCreds[0]` always be in the same order as the ones stored in the `time_limited_v2_order_creds`

Reply:
yes, they will always be in the same order, the array in sdk keeps order.